File: ProjML_ready_fixed/docker/Dockerfile.ml
FROM python:3.9-slim
WORKDIR /app
COPY ProjMLKodiim/sematic_search/semantic_search_service/requirements.txt .
RUN pip install --no-cache-dir -r requirements.txt
COPY ProjMLKodiim/sematic_search/semantic_search_service .
EXPOSE 50052
ENTRYPOINT ["python", "app.py"]

File: ProjML_ready_fixed/docker/Dockerfile.go
FROM scratch
COPY --from=builder /app/backend /backend
ENTRYPOINT ["/backend"]

FROM golang:1.23-alpine AS builder
WORKDIR /app
COPY ProjMLKodiim .
RUN go mod download
RUN CGO_ENABLED=0 GOOS=linux go build -o backend .

FROM scratch
COPY --from=builder /app/backend /backend
ENTRYPOINT ["/backend"]
